### ISSUE TITLE
label filters should not use regex match by default

### ISSUE THREAD

**Issue by damithc** [HubTurbo/HubTurbo#650]
Is it currently using a regex match? For example, does `label:p.` matches `px` label?

**Comment by dariusf** [comment]
The intended behaviour for `label:p.` is to match labels with a group that contains `p`. `priority.1`, `priority-1`, `p.x` would all be matched. `px` shouldn't be matched, since it has no group.

Is there a specific example which shows the discrepancy?

I'm currently writing tests for how label names are parsed. It's not completely consistent.

**Comment by damithc** [comment]
Why not use a simple 'startsWith' match for labels? That way, no surprises.
'p.' should not match 'priority.low' but should match 'p.low' right?

**Comment by dariusf** [comment]
Okay, that works. Should we abstract away the differences between delimiters, i.e. `p.` matches `p.1` and `p-1`?

If there is no delimiter, should the group be considered, i.e. should `label:ui` match `aspect.ui`?

**Comment by damithc** [comment]
One approach:
Use 'contains' as the default for filters and provide another way to create 'startsWith' filters e.g. `label:aspect.*` and 'endsWith' filters e.g. `label:*ui`. This approach does not require us to treat grouped labels differently from other labels. Same approach can be used for searching by keywords. i.e. we do a 'contains' search by default. When sorting by labels, 'startsWith' is the only option. `sort:aspect.*`, `sort:aspect.`, and `sort:aspect` all mean sort by the label group `aspect`. 
Any other ideas? @lejolly @acruis @rd1992

**Comment by damithc** [comment]
One drawback of taking 'contains' as the default is there is no way to specify a 'full word match'. Take 'full word match' as the default instead?

**Comment by dariusf** [comment]
Yes, with that, to express 'contains the string `abc`', the user would have to write `label:*abc*`.

`label:ui` not matching `aspect.ui` by default may be unintuitive (since only label names are shown on issue cards, without the group).

Also all other qualifiers use `contains` by default (`milestone:1` will match `V1.0`), so there's a bit of inconsistency. Then again, the other qualifiers don't have the same needs for precision.

**Comment by damithc** [comment]
Let's think about a bit more and see if there is a more consistent and intuitive approach.

**Comment by damithc** [comment]
May be use double quotes to indicate 'full word match' and 'phrase match? e.g. `"keyword"` and `"key phrase"`

**Comment by dariusf** [comment]
That could work. We're already using them to enclose strings with spaces, or other metacharacters.

**Comment by damithc** [comment]
To add on, I think we should introduce a wild card. i.e. `high` should not match `highlights` but `high*` should.

**Comment by dariusf** [comment]
The explicitness is good, but it might hamper the simple cases, e.g. we would have to type `pri*.hi*` to match `priority.high`.

**Comment by damithc** [comment]
alternatively, we can assume partial match as the default and give a way to force an exact match. Use double quotes for exact match? e.g. `label:"priority.high"` for exact match
